template/core/commons-test/src/main/java/io/bloco/commonstest/MockMainDispatcherTestRule.kt

```diff
@@ -0,0 +1,27 @@
+package io.bloco.commonstest
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.TestCoroutineScheduler
+import kotlinx.coroutines.test.TestDispatcher
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+class MockMainDispatcherTestRule(
```

Comment:
Isn't this only needed for ViewModel tests? Maybe it should be on the AndroidApp?

Reply:
watching android dev summit, there might be an alternative using 

https://github.com/googlecodelabs/kotlin-coroutines/blob/master/coroutines-codelab/finished_code/src/test/java/com/example/android/kotlincoroutines/main/utils/MainCoroutineScopeRule.kt